src/Utilities.mss

```diff
@@ -78,6 +78,72 @@ function SimpleNoteHash (nobj) {
 
 }  //$end
 
+function LayerHash (barInfo, voiceInfo) {
+    //$module(Utilities.mss)
+    // `barInfo` is either a bar number or a Bar object.
+    // `voiceInfo` is either a voice number or a BarObject from which the
+    // VoiceNumber info would be used.
+    // At least one of the two arguments must be an object.
+    //
+    // Calculates a numeric identifier from the implied bar number, voice number
+    // and staff number. The staff number is
+    // barNumber must be supplied as argument because we might either want the
+    // hash of the start layer, or the has of the end layer (where we'd supply
+    // bobj.EndBarNumber).
+    // This is called 'Layer'Hash in reference to MEI's layer element that will
+    // aggregate all elements originating from a specific staff, bar and voice.
+
+    staff = null;
+    if (IsObject(voiceInfo))
+    {
+        voiceNumber = voiceInfo.VoiceNumber;
+        staff = voiceInfo.ParentBar.ParentStaff;
+    }
+    else
+    {
+        voiceNumber = voiceInfo;
+    }
+
+    if (IsObject(barInfo))
+    {
+
+        barNumber = barInfo.BarNumber;
+        staff = barInfo.ParentStaff;
+    }
+    else
+    {
+        barNumber = barInfo;
+    }
+    if (null = staff) {
```

Comment:
This might be petty,  but it might aid the readability to not switch the agruments from the statement above...

Reply:
I adopted the habit of putting null on the left side in ManuScript because `null=CreateArray()` is false as expected, but `CreateArray()=null` is true for some reason (a general problem with TreeNodes). But comparison with null is something very hairy anyway (`0=null`, `''=null`, `' '=null`), so I'll change that.

Reply:
I was just not aware of the habit, sorry...
If you have good reasons and it works, there is no need to change that.